src/server/app/main.py

```diff
@@ -162,9 +162,16 @@ async def search_collections(
         for base_url in settings.cmr_urls
     ]
 
-    results = await search_all(executor, catalogs)
+    search_results = await search_all(executor, catalogs)
+    results = []
+    errors = []
+    for search_result in itertools.islice(search_results, limit):
+        if isinstance(search_result, CollectionMetadata):
+            results.append(search_result)
+        if isinstance(search_result, FederatedSearchError):
+            errors.append(search_result)
 
-    return SearchResponse(results=itertools.islice(results, limit))
+    return SearchResponse(results=results, errors=errors)
```

Comment:
What does the response sent back to the client look like with this change? Are you sure this doesn't break the client?

Reply:
The client is ready for this change since it reads from the `results` key in the JSON. I haven't added anything to surface the specific errors in the client application yet but most error cases should surface in the 'API Health' widget.

Reply:
Ah, I think that indirectly answers my question. I think the answer is that the client simply receives a `SearchResponse` converted to a JSON string, so with this change, the JSON now has 2 top-level keys (`"results"` and `"errors"`), whereas previously, it simply had one (`"results"`), so the change is backwards compatible.